channel_print.py

```diff
@@ -0,0 +1,314 @@
+                        raise ValueError(f"Message {i+1} invalid attach url.")
+                    _msg['file'] = msg['file']
+                result.append(_msg)
+            except TypeError:
+                raise ValueError(f"Message {i+1} not an object or string.")
+    return result
+
+
+class ChannelPrint(BasePlugin):
+    name = "channel_print"
+    version = "1.0"
+    author = "GTG3000"
+    description = "A plugin that allows printing of prepared multi-message data."
+
+    default_config = {
+        "max_filesize": 1024 * 1024 * 8  # max 8 mb
+    }
+
+    walls: JsonFileDict
+
+    async def activate(self):
+        self.walls = self.config_manager.get_plugin_config_file("walls.json",
+                                                                json_save_args={'indent': 2, 'ensure_ascii': False})
+
+    @Command("Print",
+             doc="Prints out the specified document from the storage, allowing to dump large amounts of information "
+                 "into a channel, for example for purposes of a rules channel.\n"
+                 "Document can be specified from the saved ones, or uploaded with the command.",
+             syntax="(document)",
+             perms={"manage_messages"},
+             delcall=True)
+    async def _print(self, msg):
+        gid = str(msg.guild.id)
+
+        if msg.attachments:
+            _file = BytesIO()
+
+            await msg.attachments[0].save(_file)
+            try:
+                wall = decode_json(_file.getvalue())
+            except ValueError as e:
+                self.logger.exception("Could not decode uploaded document file!", exc_info=True)
+                raise CommandSyntaxError(e)
+            except Exception as e:
+                raise CommandSyntaxError(f"Not a valid JSON file: {e}")
+
+            try:
+                wall = verify_document(wall)
+            except ValueError as e:
+                raise CommandSyntaxError(e)
+        else:
+            if gid not in self.walls:
+                self.walls[gid] = dict()
+                return
+
+            try:
+                wall = msg.clean_content.split(None, 1)[1]
+            except IndexError:
+                raise CommandSyntaxError
+
+            if wall not in self.walls[gid]:
+                raise CommandSyntaxError("No such document.")
+
+            try:
+                wall = verify_document(self.walls[gid][wall])
+            except ValueError as e:
+                await respond(msg, f"**WARNING: {e}**")
+                return
+
+        for post in wall:
+            if isinstance(post, str):
+                await respond(msg, post)
+            else:
+                _file = None
+                if 'file' in post:
+                    try:
+                        _file = urlopen(post['file'])
+                        if int(_file.info()['Content-Length']) > self.plugin_config['max_filesize']:
+                            raise ValueError("File too big.")
```

Comment:
This should probably be a CommandSyntaxError or UserPermissionError

Reply:
This is being caught by the `except` right after it. If the file is unfit, it will fail quietly and print about it via the logger.

Perhaps a better idea would be to add a file size verification before hand. For some reason, I felt like verifying file links ahead of time could not be a great idea, but I suppose it's on the users for changing the contents?